--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,6 +74,15 @@
 ##################  some compiler settings ################################
 if(CMAKE_COMPILER_IS_GNUCXX AND NOT WIN32 )
     set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wnon-virtual-dtor -Wno-long-long -ansi -Wundef -Wcast-align -Wchar-subscripts -Wall -W -Wpointer-arith -Wformat-security -fno-check-new -fno-common")
+    include(CheckCXXCompilerFlag)
+    check_cxx_compiler_flag(-fvisibility=hidden __SOPRANO_HAVE_GCC_VISIBILITY)
+    check_cxx_compiler_flag(-fvisibility-inlines-hidden __SOPRANO_HAVE_GCC_INLINE_VISIBILITY)
+    if(__SOPRANO_HAVE_GCC_VISIBILITY)
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=hidden")
+    endif(__SOPRANO_HAVE_GCC_VISIBILITY)
+    if(__SOPRANO_HAVE_GCC_INLINE_VISIBILITY)
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility-inlines-hidden")
+    endif(__SOPRANO_HAVE_GCC_INLINE_VISIBILITY)
 endif(CMAKE_COMPILER_IS_GNUCXX AND NOT WIN32 )
 if(MSVC)
     add_definitions(-D_CRT_SECURE_NO_DEPRECATE -D_CRT_NONSTDC_NO_DEPRECATE -Zc:wchar_t-)
--- a/index/CMakeLists.txt
+++ b/index/CMakeLists.txt
@@ -27,15 +27,14 @@
 # CLucene requires exception support and has no support for visibility=hidden
 # so we must use the default (i.e. public) value for -fvisibility
 IF(NOT WIN32)
-    include(CheckCXXCompilerFlag)
-    # Only GCC supports these flags
     IF(CMAKE_COMPILER_IS_GNUCXX)
-        check_cxx_compiler_flag(-fvisibility=hidden __SOPRANO_HAVE_GCC_VISIBILITY)
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fexceptions")
         if(__SOPRANO_HAVE_GCC_VISIBILITY)
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fexceptions -fvisibility=default")
-        else(__SOPRANO_HAVE_GCC_VISIBILITY)
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fexceptions")
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=default")
         endif(__SOPRANO_HAVE_GCC_VISIBILITY)
+        if(__SOPRANO_HAVE_GCC_INLINE_VISIBILITY)
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-visibility-inlines-hidden")
+        endif(__SOPRANO_HAVE_GCC_INLINE_VISIBILITY)
     ENDIF(CMAKE_COMPILER_IS_GNUCXX)
 ENDIF(NOT WIN32)
 
