Index: soprano-1.99~rc2/CMakeLists.txt
===================================================================
--- soprano-1.99~rc2.orig/CMakeLists.txt	2007-12-16 19:17:18.000000000 +0100
+++ soprano-1.99~rc2/CMakeLists.txt	2007-12-16 19:17:40.000000000 +0100
@@ -17,19 +17,6 @@
 
 find_package(Redland)
 
-find_package(JNI)
-if(JAVA_INCLUDE_PATH AND JAVA_JVM_LIBRARY)
-  file(READ ${JAVA_INCLUDE_PATH}/jni.h jni_header_data)
-  string(REGEX MATCH "JNI_VERSION_1_6" JNI_1_6_FOUND "${jni_header_data}")
-  if(JNI_1_6_FOUND)
-    message(STATUS "Found Java JNI >= 1.6: ${JAVA_INCLUDE_PATH}, ${JAVA_JVM_LIBRARY}")
-  else(JNI_1_6_FOUND)
-    message( "Need JRE version 1.6 or higher for the Sesame2 backend.")
-  endif(JNI_1_6_FOUND)
-else(JAVA_INCLUDE_PATH AND JAVA_JVM_LIBRARY)
-  message(STATUS "Could not find Java JNI")
-endif(JAVA_INCLUDE_PATH AND JAVA_JVM_LIBRARY)
-
 set (LIB_SUFFIX "" CACHE STRING "Define suffix of directory name (32/64)" )
 set (LIB_DESTINATION "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}" CACHE STRING "Library directory name" FORCE)
 ## the following are directories where stuff will be installed to
@@ -95,16 +82,13 @@
 
 message("---------------------------------------------------------------------------------------")
 message("-- Soprano Components that will be built:")
-if(NOT REDLAND_FOUND AND JNI_1_6_FOUND)
+if(NOT REDLAND_FOUND)
   message("   WARNING: No Soprano backends will be compiled due to missing dependancies!")
   message("   WARNING: Soprano will not be very useful without storage backends!")
-endif(NOT REDLAND_FOUND AND JNI_1_6_FOUND)
+endif(NOT REDLAND_FOUND)
 if(REDLAND_FOUND)
   message("   * Redland storage backend")
 endif(REDLAND_FOUND)
-if(JNI_1_6_FOUND)
-  message("   * Sesame2 storage backend (java-based)")
-endif(JNI_1_6_FOUND)
 
 if(REDLAND_FOUND)
   message("   * Raptor RDF parser")
@@ -127,9 +111,6 @@
 if(NOT REDLAND_FOUND)
   message("   * Redland storage backend")
 endif(NOT REDLAND_FOUND)
-if(NOT JNI_1_6_FOUND)
-  message("   * Sesame2 storage backend (java-based)")
-endif(NOT JNI_1_6_FOUND)
 
 if(NOT REDLAND_FOUND)
   message("   * Raptor RDF parser")
Index: soprano-1.99~rc2/backends/CMakeLists.txt
===================================================================
--- soprano-1.99~rc2.orig/backends/CMakeLists.txt	2007-12-16 19:17:18.000000000 +0100
+++ soprano-1.99~rc2/backends/CMakeLists.txt	2007-12-16 19:17:40.000000000 +0100
@@ -4,6 +4,3 @@
   add_subdirectory(redland)
 endif(REDLAND_FOUND)
 
-if(JNI_1_6_FOUND)
-  add_subdirectory(sesame2)
-endif(JNI_1_6_FOUND)
