commit ae27fd59085fdddc1a167c2ce35f5be52e75ef35
Author: Maximiliano Curia <maxy@debian.org>
Date:   Fri Jun 7 12:36:07 2013 +0200

    Add m-a dirs to the search path.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5f96c97..173a864 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -143,6 +143,21 @@ set (LIB_DESTINATION "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}" CACHE STRING "Li
 set(INCLUDE_INSTALL_DIR      "${CMAKE_INSTALL_PREFIX}/include/" CACHE PATH "The subdirectory to the header prefix")
 set(PKGCONFIG_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/pkgconfig/" CACHE STRING "Base directory for pkgconfig files")
 
+find_program (
+  DPKG_ARCHITECTURE
+  NAMES dpkg-architecture
+  )
+if (DPKG_ARCHITECTURE)
+    execute_process (
+        COMMAND ${DPKG_ARCHITECTURE} -qDEB_HOST_MULTIARCH
+        OUTPUT_VARIABLE DEB_HOST_MULTIARCH
+    )
+    if (DEB_HOST_MULTIARCH)
+        string(REPLACE "\n" "" DEB_HOST_MULTIARCH ${DEB_HOST_MULTIARCH})
+        add_definitions(-DDEB_HOST_MULTIARCH="${DEB_HOST_MULTIARCH}")
+    endif (DEB_HOST_MULTIARCH)
+endif (DPKG_ARCHITECTURE)
+
 
 # Set up RPATH handling, so the libs are found if they are installed to a non-standard location.
 # By default cmake builds the targets with full RPATH to everything in the build directory,
diff --git a/soprano/sopranodirs.cpp b/soprano/sopranodirs.cpp
index e6292af..9f302b1 100644
--- a/soprano/sopranodirs.cpp
+++ b/soprano/sopranodirs.cpp
@@ -132,6 +132,10 @@ QStringList Soprano::libDirs()
 #else
     paths << QLatin1String( "/usr/lib"SOPRANO_LIB_SUFFIX );
     paths << QLatin1String( "/usr/local/lib"SOPRANO_LIB_SUFFIX );
+#ifdef DEB_HOST_MULTIARCH
+    paths << QLatin1String( "/usr/lib"SOPRANO_LIB_SUFFIX"/"DEB_HOST_MULTIARCH );
+    paths << QLatin1String( "/usr/local/lib/"SOPRANO_LIB_SUFFIX"/"DEB_HOST_MULTIARCH );
+#endif
     paths += Soprano::envDirList( "LD_LIBRARY_PATH" );
 #endif
     return paths;
